商品出品機能

miracletea opened the pull request:
# What
商品出品機能を作成しました。

# Why
商品出品機能の実装のため。

【Gyazo】
・ログイン状態の場合は、商品出品ページへ遷移できる動画
https://gyazo.com/616cdfe27de9e983399b7adce3d2bdb0

 ・価格が入力されると同時に、販売手数料と販売利益が表示される動画
https://gyazo.com/24df7a9fe873da824854ede6b5d9fbc0

 ・必要な情報を適切に入力して「出品する」ボタンを押すと、商品情報がデータベースに保存される動画
https://gyazo.com/108641d642e0fdee7104f38e45faa892

 ・入力に問題がある状態で「出品する」ボタンが押された場合、情報は保存されず、出品ページに戻りエラーメッセージが表示される動画
https://gyazo.com/142c27aa133fa1e9b7482a614617e580

・ ログアウト状態の場合は、商品出品ページへ遷移しようとすると、ログインページへ遷移する動画
https://gyazo.com/30ea83a7d711d8d9d2fd17653bd89e86

 ・テスト結果の画像
https://gyazo.com/2ab2364cc9b6bd37f79c1df4474d5b95

miracletea committed:
itemモデルとActivesorageの導入

miracletea committed:
商品の情報入力の実装

miracletea committed:
販売手数料・利益計算のjavascript実装

miracletea committed:
エラーメッセージの表示

miracletea committed:
ログアウト時のページ遷移追加

miracletea committed:
商品出品機能のテストコード実装

koyamatakumi commented on app/models/item.rb:
```diff
@@ -0,0 +1,28 @@
+
+  validate :image_attached
+  validates :name, :detail,presence: true
+  validates :price, presence: true, numericality: { greater_than_or_equal_to: 300, less_than_or_equal_to: 9999999 }
```
```ruby
  validates :price, presence: true, numericality: { greater_than_or_equal_to: 300, less_than_or_equal_to: 9999999 }
```
only_integer:trueのオプションを追加しましょう。

koyamatakumi replied:
[こちら](https://github.com/miracletea/furima-39667/pull/14/files#r1435504127)の内容が修正されておりません。
整数のみが保存されるように、only_integer:trueのオプションを追加しましょう。

koyamatakumi commented on app/views/items/new.html.erb:
```diff
@@ -0,0 +1,159 @@
+
+    <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
+    <%= render 'shared/error_messages', model: f.object %>
+    <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
```
```ruby
    <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>

    <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
```
実装完了後は実装案内のコメントは削除しておきましょう！

koyamatakumi commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,104 @@
+        expect(@item).to be_valid
+      end
+    end
+    context '商品出品できないとき' do
```
以下のテストを追加しましょう。
```
userが紐づいていない場合登録できない。
```

koyamatakumi reviewed (commented):
miracletea様レビュー依頼フォームの送信ありがとうございます。
内容を確認しコメントをさせていただきました。
修正いただき改めてレビュー依頼フォームの送信をお願いいたします。
引き続き頑張っていきましょう！

miracletea committed:
only_integer:trueのオプションの追加

app/models/item.rb

  validate :image_attached
  validates :name, :detail,presence: true
  validates :price, presence: true, numericality: { greater_than_or_equal_to: 300, less_than_or_equal_to: 9999999 }
@koyamatakumi koyamatakumi 7 minutes ago
  validates :price, presence: true, numericality: { greater_than_or_equal_to: 300, less_than_or_equal_to: 9999999 }
only_integer:trueのオプションを追加しましょう。

miracletea committed:
不要コメントを削除

app/views/items/new.html.erb
    <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
    <%= render 'shared/error_messages', model: f.object %>
    <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
@koyamatakumi koyamatakumi 8 minutes ago
    <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>

    <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
実装完了後は実装案内のコメントは削除しておきましょう！

miracletea committed:
only_integer: trueの削除と追加

only_integer: trueを誤ってFakerに記述していたため削除。itemモデルのvalidatesに追加

miracletea commented:
ご確認ありがとうございます。
コメント反映いたしました、宜しくお願い致します。

【Gyazo】
テスト追加後の結果
https://gyazo.com/1ccde485fd54931f31996ac842785801

koyamatakumi reviewed (commented):
miracletea様レビュー依頼フォームの送信ありがとうございます。
内容を確認しコメントをさせていただきました。
修正いただき改めてレビュー依頼フォームの送信をお願いいたします。
あと少しです、頑張っていきましょう！

miracletea committed:
only_integer: trueの見直し

priceにオプションを付ける

miracletea commented:
修正ミスしてました、すみません。
再度ご確認お願いいたします。

【Gyazo】
テスト追加後の結果
https://gyazo.com/6e08611434b6695ea9f09b4647e6c758

koyamatakumi reviewed (commented):
miracletea様レビュー依頼フォームの送信ありがとうございます。
内容を確認させていただき問題ないことが確認できました。
これにてLGTMとさせていただきます。

マージとプルを忘れずに行い、次の実装に進んで参りましょう。
引き続き応援しております！
